core/services/gateway/handlers/common/callback.go

```diff
@@ -0,0 +1,46 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/handlers"
+)
+
+type Callback struct {
+	c    chan handlers.UserCallbackPayload
+	mu   sync.Mutex
+	sent bool
+}
+
+func (c *Callback) SendResponse(ctx context.Context, payload handlers.UserCallbackPayload) error {
+	c.mu.Lock()
```

Comment:
nit: You could use this and avoid a mutex: https://pkg.go.dev/sync/atomic#Bool.CompareAndSwap